pkg/apis/v1alpha3/types.go

```diff
@@ -49,12 +49,13 @@ const (
 
 // LogStatus defines the current status of the log resource.
 type LogStatus struct {
-	Path string `json:"path,omitempty"`
-	Size int64  `json:"size"`
+	Path     string `json:"path,omitempty"`
+	Size     int64  `json:"size"`
+	IsStored bool   `json:"isStored"`
```

Comment:
let's add an `errorOnStoreCode int `json:"errorOnStore"` field and an `errorOnStoreMsg string `json:"errorOnStoreMsg"`  field

that way we can tell if 

- an attempt to store has not yet completed
- the store was successful
- the store failed on an error
- the precise error is retryable or not

Reply:
Yes, that seems a better idea. Should I replace `IsStored` or add these new fields?

Reply:
let's keep IsStored and add the other two

Reply:
What's `errorOnStoreCode int`?

Reply:
GRPC has a standard set of error codes per https://grpc.github.io/grpc/core/md_doc_statuscodes.html

In fact, when you make metric queries like 
```
grpc_server_handled_total{job="tekton-results-api", grpc_service=~"tekton.results.v1alpha2.*", grpc_method='UpdateLog'}
```

you'll see a grpc_code column with each of those status codes listed

I envision potentially gating retry on log storage errors based on those error codes, similar to what we do in k8s controllers @khrm